rtengine/rt_math.h

```diff
@@ -101,5 +101,17 @@ inline T norminf(const T& x, const T& y)
     return std::max(std::abs(x), std::abs(y));
 }
 
+inline int float2uint16range(float d) // clips input to [0;65535] and rounds
+{
+    d = CLIP(d); // clip to [0;65535]
+#ifdef __SSE2__ // this only works in IEEE 754 maths. For simplicity I restricted it to SSE2. We can enhance it later, but we have to take care of endianness then.
+    d += 12582912.f;
+    return reinterpret_cast<int&>(d);
```

Comment:
@heckflosse Great adaption of the Lua double code! I was wondering, why you cast to `int&` here instead of `int`. What's the reason here?